solidity/ecdsa/contracts/WalletRegistry.sol

```diff
@@ -99,8 +104,8 @@ contract WalletRegistry is
 
     // External dependencies
 
-    SortitionPool public immutable sortitionPool;
-    IStaking public immutable staking;
+    SortitionPool public sortitionPool;
+    IStaking public staking;
```

Comment:
If you want, you can still use immutable variables, as well as using a constructor to initialize them. See the `constructorArgs` option in https://docs.openzeppelin.com/upgrades-plugins/1.x/api-hardhat-upgrades

Reply:
Thanks for the comment! Initially, I was thinking that using both constructor and initializer will be too confusing, so I decided to use just one. But to be honest upgradable contracts are already complex and confusing so this constructor+initializer won't add much complexity 😉 
I updated the code in 35d49261b56857c35df4830cb7f6cc7de11d77f0